doc/user/metrics/README.md

```diff
@@ -74,15 +74,129 @@ In Kubernetes clusters where [OwnerReferencesPermissionEnforcement][ownerref-per
 
 By default operator will expose info metrics based on the number of the current instances of an operator's custom resources in the cluster. It leverages [kube-state-metrics][ksm] as a library to generate those metrics. Metrics initialization lives in the `cmd/manager/main.go` file of the operator in the `serveCRMetrics` function. Its arguments are a custom resource's group, version, and kind to generate the metrics. The metrics are served on `0.0.0.0:8686/metrics` by default. To modify the exposed metrics port number, change the `operatorMetricsPort` variable at the top of the `cmd/manager/main.go` file in the generated operator.
 
+### Custom Metrics using Operator-SDK
+To implement custom metric collection using the `metrics package` provided by SDK, follow the steps outlined below:
+1. Update the Operator: Update the operator to provide sufficient permisisons so that the required resources for prometheus can be created.
+2. Custom metric collection: Implement the logic for custom metric variables defined for your operator and register the specific metrics with prometheus. 
+3. Expose metrics: Utilize the functions provided by SDK to expose metrics and start the metrics server.
+
+#### Update the Operator
+
+Update the operator and provide sufficient permissions to allow creation of Prometheus resources. The required permissions are:
+1. `Service Monitor` : `get` and `create`
+2. `roles` and `rolebindings` : `get`, `create`, `delete`, `update`, `list`
+
+The below is a snippet from roles.yaml:
+```yaml
+- apiGroups:
+  - rbac.authorization.k8s.io
+  resources:
+  - roles
+  - rolebindings
+  verbs:
+  - get
+  - create
+  - delete
+  - update
+  - list
+  - watch
+- apiGroups:
+  - monitoring.coreos.com
+  resources:
+  - servicemonitors
+  verbs:
+  - get
+  - create
+
+```
+#### Custom metric collection
+
+Implement the logic to collect specific custom metrics inside the operator code base. For example, the following is a snippet from `Marketplace` operator which implements custom metric collection:
+
+```Go
+// Metric variables and metric list that are used to report error. 
+var (
+	errorMetric = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "error_count",
+			Help: "Describes the monotonic count of the error",
+		},
+		[]string{errorName, errorValue},
+	)
+    metricsList = []prometheus.Collector{
+		errorMetric,
+	}
+)
+```
+Register the metrics created above with Prometheus:
+
+```Go
+// registerMetrics registers the metrics with prometheus.
+func registerMetrics() error {
+	for _, metric := range metricsList {
+		err := prometheus.Register(metric)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+```
+#### Expose metrics
+The steps provided previously for creating a service and service monitor resource can be followed to expose custom metrics. Additionally, the metrics server has to be manually started instead of providing it as an argument to the controller-runtime manager.
+Usage:
+
```

Comment:
Does NOT the following code generated by default in the scalfod project? WDYT about adding an example of a customized metric above using the Memcached sample and them show here just what needs to be done to expose it?

Reply:
Yes, the code is now generated by default in the scaffold project. But the server has to still be started again. I have provided the code just as an example to do so.

Reply:
It shows fine.